kuiper-game/Domain/Ship.cs

```diff
@@ -48,5 +52,83 @@ public Ship(string name, string shipClass, int speed)
                 return $"{Name} {ShipStatusExtensions.ToFriendlyString(Status)} {CurrentLocation.Name}";
             }
         }
+
+        public string SetCourse(CelestialBody target)
+        {
+            if(target == CurrentLocation)
+            {
+                return $"{Name} is already in orbit above {CurrentLocation.Name}";
+            }
+            if(target == TargetLocation)
+            {
+                return $"{Name} is already enroute to {TargetLocation.Name}";
+            }
+            Status = ShipStatus.Enroute;
+            TargetLocation = target;
+            long distance = SolarSystemLocator.SolarSystem.SolarSystemService.GetDistanceInKm(CurrentLocation, TargetLocation);
+
+            var hoursToTargetLocation = distance/Speed;
+            ArrivalTime = GameTime.Now() + TimeSpan.FromHours(hoursToTargetLocation);
+            var evt = new CourseSet(GameTime.Now(), TimeSpan.FromHours(hoursToTargetLocation));
+            Enqueue(evt);
+            return evt.StartEvent();
+        }
+
+        public string ScanForAsteroids()
+        {
+            if(Status == ShipStatus.InOrbit)
+            {
+                var evt = new MiningScan(GameTime.Now());
+                Enqueue(evt);
+                return evt.StartEvent();
+            }
+            return $"Ship must be in orbit around a stabile location to scan for asteroids";
+        }
+
+        public string MineAsteroid(int durationHours)
+        {
+            if(Status == ShipStatus.InOrbit && CurrentLocation is Asteroid)
```

Comment:
@Nicoolai is this the right way to check if the current nearest body is the right type?

Like if we had a "land" action, that would check if it wasn't a Gas Giant or something like that.

Reply:
Yea. 
I'm trying to think which entity should be aware of something like that. I guess it makes sense that it is the ship that knows what it can do, to various types of CelestialBodies, instead of the actual bodies.

Maybe there's a special ship-type, at some point, that can enter a gasgiant and do something special.